src/main.rs

```diff
@@ -339,6 +382,30 @@ fn cargo_careful(args: env::Args) {
         cmd.args(["--target", target.as_str()]);
     }
 
+    // Enable Main Thread Checker on Apple platforms, as documented here:
+    // <https://developer.apple.com/documentation/xcode/diagnosing-memory-thread-and-crash-issues-early#Detect-improper-UI-updates-on-background-threads>
+    //
+    // `DYLD_INSERT_LIBRARIES` is not inherited by child processes, so we need
+    // to tell Cargo to set this environment variable for the processes it
+    // launches (instead of just setting it using `cmd.env`).
```

Comment:
That's pretty surprising, are you sure there's not some other weirdness going on? Are there docs saying that DYLD_INSERT_LIBRARIES indeed only goes "one process deep"?

Reply:
Huh, my apologies, it's something that the main thread checker itself does, and which can be controlled with [`MTC_RESET_INSERT_LIBRARIES`](https://bryce.co/main-thread-checker-configuration/#mtc_reset_insert_libraries).

Do you want me to use that flag to disable this behaviour, and then just go with the usual `cmd.env`?

Reply:
> Huh, my apologies, it's something that the main thread checker itself does, and which can be controlled with [MTC_RESET_INSERT_LIBRARIES](https://bryce.co/main-thread-checker-configuration/#mtc_reset_insert_libraries).

Ah I see, makes sense. Please add a link to that.